src/plugins/vctlGit.nim

```diff
@@ -792,16 +792,22 @@ proc isInRepo(obj: ChalkObj, repo: string): bool =
 
   return false
 
-proc gitInit(self: Plugin) =
-  let cache = GitInfo(self.internalState)
+proc gitInit*(self: Plugin): Plugin =
+  once:
+    let cache = GitInfo(self.internalState)
+    for path in getContextDirectories():
+      cache.findAndLoad(path.resolvePath())
+  return self
 
-  for path in getContextDirectories():
-    cache.findAndLoad(path.resolvePath())
+proc gitFirstDir*(self: Plugin): string =
```

Comment:
I know we paired on this approach together but thinking about it a little perhaps better to add `gitInit` inside this function so taht the caller doesnt need to worry about if git plugin was initialized or not. all publicly exported methods will do that directly so the usage is simpler? thoughts?

Reply:
So the convention is, all public funcs from the vctlGit plugin should start with something like:

```nim
# ...etc...
let cache = self.GitInit.internal_state
# ...etc...
```

Do we want to make that consistent in another PR for all plugins? For example `procfs` exposes `procfsGetRunTimeHostInfo`. It could instead grow a `ProcFsInit` similar to `GitInit`.

```nim
proc procfsGetRunTimeHostInfo*(self: Plugin, objs: seq[ChalkObj]):
       ChalkDict {.cdecl.} =
  result    = ChalkDict()
  let cache = ProcFsCache(self.internalState)
  # ...etc...
```

Reply:
not following your comment

no changes necessary in any of the plugins. Im just saying we can move the initialization logic to be completely private to the plugin:

```nim
proc gitInit(self: Plugin) = # note not exported functions as ther is no *
  ...

proc gitFirstDir*(self: Plugin): string =
  self.gitInit()
  ...
```

then the caller is just:

```nim
getPluginByName("vctl_git").gitFirstDir()
```